src/box/lua/net_box.lua

```diff
@@ -772,7 +816,9 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
-    return (pcall(self._request, self, M_PING, opts, nil, self._stream_id))
+    check_param_table(opts, REQUEST_OPTION_TYPES)
```

Comment:
You check the options, but still allow `is_async` which is now not fixed by any commit. Hence it is invalid for ping. But I can see that it works and even returns `true`. Lets ban it as an invalid option.
```
tarantool> c:ping({is_async = true})
---
- true
...
```

Reply:
Done. Moreover, as we check that opts have a table type in `_request` and non-supported options are checked inside 'wrappers', we should also add `type(opts) == 'table'` in such checks (non-supported ones) as otherwise error won't be very informative: 
```lua
> box.space.test.index.primary:get({1}, {buffer = 1})
error: builtin/box/net_box.lua:1098: attempt to index local 'opts' (a number value)
```